c/shared/source/ufowrite/ufowrite.c

```diff
@@ -650,6 +553,219 @@ static void writeBlueValues(ufwCtx h, abfPrivateDict *privateDict){
+            fd->FontMatrix.array[1] = 0.0;
+            fd->FontMatrix.array[2] = 0.0;
+            fd->FontMatrix.array[3] = 0.001;
+            fd->FontMatrix.array[4] = 0.0;
+            fd->FontMatrix.array[5] = 0.0;
+        }
+        writeLine(h, "\t<key>FontMatrix</key>");
+        writeLine(h, "\t<array>");
+        for (i = 0; i < fd->FontMatrix.cnt; i++) {
+            float stem = fd->FontMatrix.array[i];
+            if (stem == ((int)stem))
+                sprintf(buffer, "\t\t<integer>%d</integer>", (int)stem);
+            else
+                sprintf(buffer, "\t\t<real>%.3f</real>", stem);
+            writeLine(h, buffer);
+        }
+        writeLine(h, "\t</array>");
+
+        privateDict = &(fd->Private);
+        writeLine(h, "\t<key>PrivateDict</key>");
+        writeLine(h, "\t<dict>");
+        writeBlueValues(h, privateDict);
+        writeLine(h, "\t</dict>");
+        writeLine(h, "\t</dict>");
+    }
+    writeLine(h, "\t</array>");
+}
+
+static void writeCIDMap(ufwCtx h, abfTopDict *top, char *buffer) {
+    int i;
+    writeLine(h, "\t<key>com.adobe.type.postscriptCIDMap</key>");
+    writeLine(h, "\t<dict>");
+    for (i = 0; i < h->glyphs.cnt; i++) {
+        sprintf(buffer, "\t\t<key>%s</key>", h->glyphs.array[i].glyphName);
+        writeLine(h, buffer);
+        sprintf(buffer, "\t\t<integer>%d</integer>", h->glyphs.array[i].cid);
+        writeLine(h, buffer);
+    }
+    writeLine(h, "\t</dict>");
+}
+
+static void writeGlyphOrder(ufwCtx h) {
+    char buffer[FILENAME_MAX];
+    int i;
+
+    /* Set error handler */
+    DURING_EX(h->err.env)
+
+    h->state = 1; /* Indicates writing to dst stream */
+
+    /* Open lib.plist file as dst stream */
+
+    sprintf(buffer, "%s", "lib.plist");
+    h->cb.stm.clientFileName = buffer;
+    h->stm.dst = h->cb.stm.open(&h->cb.stm, UFW_DST_STREAM_ID, 0);
+    if (h->stm.dst == NULL)
+        fatal(h, ufwErrDstStream);
+
+    writeLine(h, XML_HEADER);
+    writeLine(h, PLIST_DTD_HEADER);
+    writeLine(h, "<plist version=\"1.0\">");
+    writeLine(h, "<dict>");
+    writeLine(h, "\t<key>public.glyphOrder</key>");
+    writeLine(h, "\t<array>");
+    for (i = 0; i < h->glyphs.cnt; i++) {
+        Glyph *glyphRec;
+        glyphRec = &h->glyphs.array[i];
+        sprintf(buffer, "\t\t<string>%s</string>", glyphRec->glyphName);
+        writeLine(h, buffer);
+    }
+
+    writeLine(h, "\t</array>");
+
+    if (h->top->sup.flags & ABF_CID_FONT) {
+        if (h->top->cid.CIDFontName.ptr != NULL) {
+            writeLine(h, "\t<key>com.adobe.type.CIDFontName</key>");
+            sprintf(buffer, "\t<string>%s</string>", h->top->cid.CIDFontName.ptr);
+            writeLine(h, buffer);
+            writeLine(h, "\t<key>com.adobe.type.ROS</key>");
+            sprintf(buffer, "\t<string>%s-%s-%ld</string>", h->top->cid.Registry.ptr, h->top->cid.Ordering.ptr, h->top->cid.Supplement);
+            writeLine(h, buffer);
+        }
+        writeFDArray(h, h->top, buffer);
```

Comment:
What if there's an FDArray in a non-CID font?

Reply:
[Similarly to the comment here](https://github.com/adobe-type-tools/afdko/pull/1576#discussion_r1109234784), I think we can support FDArrays in non-CID fonts in an upcoming PR — let me know if you have thoughts!